witness/src/Dscp/Witness/Launcher/Runner.hs

```diff
@@ -5,22 +5,34 @@ module Dscp.Witness.Launcher.Runner
     , launchWitnessRealMode
     ) where
 
+import qualified Control.Concurrent.STM as STM
+
 import Dscp.Launcher.Rio (runRIO)
 import Dscp.Resource.Class (AllocResource (..), InitParams (..))
 import Dscp.Resource.Functions (runResourceAllocation)
 import Dscp.Snowdrop.Actions (initSDActions)
 import Dscp.Witness.Config
-import Dscp.Witness.Launcher.Mode (WitnessContext (..), WitnessRealMode)
+import Dscp.Witness.Launcher.Mode (TxRelayInput (..), TxRelayPipe (..), WitnessContext (..),
+                                   WitnessRealMode)
 import Dscp.Witness.Launcher.Params (WitnessParams (..))
 import Dscp.Witness.Launcher.Resource (WitnessResources (..))
 import Dscp.Witness.Mempool (newMempoolVar)
 
 -- | Make up Witness context from dedicated pack of allocated resources.
 formWitnessContext :: WitnessParams -> WitnessResources -> IO WitnessContext
 formWitnessContext _wcParams _wcResources = do
-    _wcMempool <- newMempoolVar
-    _wcSDActions <- initSDActions
-    pure $ WitnessContext {..}
+    _wcMempool      <- newMempoolVar
+    _wcSDActions    <- initSDActions
+    _wcTxRelayInput <- TxRelayInput <$> atomically STM.newTQueue
+    _wcTxRelayPipe  <- TxRelayPipe  <$> atomically STM.newTQueue
+    pure $ WitnessContext
+        { _wcParams
+        , _wcResources
+        , _wcMempool
+        , _wcSDActions
+        , _wcTxRelayInput
+        , _wcTxRelayPipe
+        }
```

Comment:
Wow, didn't know such syntax is possible

Reply:
`NamedRecordPuns`